velox/exec/tests/AssignUniqueIdTest.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/tests/OperatorTestBase.h"
+#include "velox/exec/tests/PlanBuilder.h"
+#include "velox/exec/tests/QueryAssertions.h"
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+using functions::test::FunctionBaseTest;
+
+class AssignUniqueIdTest : public OperatorTestBase {
+ protected:
+  void assertUniqueIdAdded(
+      const std::shared_ptr<const core::PlanNode>& plan,
+      const std::vector<RowVectorPtr> inputRows) {
+    CursorParameters params;
+    params.planNode = plan;
+    auto result = readCursor(params, [](auto /*task*/) {});
+    auto numColumns = result.second[0]->childrenSize();
+    ASSERT_EQ(numColumns, inputRows[0]->childrenSize() + 1);
+
+    for (int i = 0; i < numColumns; i++) {
+      auto column = result.second[0]->childAt(i);
+      if (i < numColumns - 1) {
+        FunctionBaseTest::assertEqualVectors(inputRows[0]->childAt(i), column);
+      } else {
+        std::set<int64_t> ids;
+        auto idValues = column->asFlatVector<int64_t>()->rawValues();
+        for (int j = 0; j < column->size(); j++) {
+          ids.insert(idValues[j]);
+        }
+        ASSERT_EQ(column->size(), ids.size());
+      }
+    }
+  }
+};
+
+TEST_F(AssignUniqueIdTest, basic) {
+  auto row = makeRowVector(
```

Comment:
Can we use a bit more data here? Let's say make this a vector of 1K rows and make a few of these?

```
makeFlatVector<int32_t>(size, [] (auto row) { return row; });
```